From a5266b4377f4ec9f146fc9613c5c9ea8a2108e09 Mon Sep 17 00:00:00 2001
From: mbykhovtsev-ms <mbykhovtsev@microsoft.com>
Date: Thu, 23 Mar 2023 15:30:31 -0700
Subject: [PATCH 2/6] Invoke logback via reflection

---
 .../cli/logging/impl/LogbackConfiguration.java | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/maven-embedder/src/main/java/org/apache/maven/cli/logging/impl/LogbackConfiguration.java b/maven-embedder/src/main/java/org/apache/maven/cli/logging/impl/LogbackConfiguration.java
index d16eaa9..05a511a 100644
--- a/maven-embedder/src/main/java/org/apache/maven/cli/logging/impl/LogbackConfiguration.java
+++ b/maven-embedder/src/main/java/org/apache/maven/cli/logging/impl/LogbackConfiguration.java
@@ -36,21 +36,31 @@ public class LogbackConfiguration
     public void setRootLoggerLevel( Level level )
     {
         ch.qos.logback.classic.Level value;
+        String value;
         switch ( level )
         {
             case DEBUG:
-                value = ch.qos.logback.classic.Level.DEBUG;
+                value = "DEBUG";
                 break;
 
             case INFO:
-                value = ch.qos.logback.classic.Level.INFO;
+                value = "INFO";
                 break;
 
             default:
-                value = ch.qos.logback.classic.Level.ERROR;
+                value = "ERROR";
                 break;
         }
-        ( (ch.qos.logback.classic.Logger) LoggerFactory.getLogger( Logger.ROOT_LOGGER_NAME ) ).setLevel( value );
+        Logger logger = LoggerFactory.getLogger( Logger.ROOT_LOGGER_NAME );
+        try {
+            Class<?> levelClass = Class.forName("ch.qos.logback.classic.Level");
+            Object logbackLevel = levelClass.getField(value).get(null);
+            Class<?> loggerClass = Class.forName("ch.qos.logback.classic.Logger");
+            loggerClass.getMethod("setLevel", new Class<?>[] {levelClass})
+                .invoke(logger, new Object[] {logbackLevel});
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to initialize logback configuration", e);
+        }
     }
 
     @Override
-- 
2.25.1

