From b1dffe796118200ae842829b5144f31756392efa Mon Sep 17 00:00:00 2001
From: mbykhovtsev-ms <mbykhovtsev@microsoft.com>
Date: Thu, 23 Mar 2023 15:35:48 -0700
Subject: [PATCH 4/6] Remove dependency on powermock

---
 .../StringSearchModelInterpolatorTest.java    | 66 -------------------
 1 file changed, 66 deletions(-)

diff --git a/maven-model-builder/src/test/java/org/apache/maven/model/interpolation/StringSearchModelInterpolatorTest.java b/maven-model-builder/src/test/java/org/apache/maven/model/interpolation/StringSearchModelInterpolatorTest.java
index 4e638e1..dc5b097 100644
--- a/maven-model-builder/src/test/java/org/apache/maven/model/interpolation/StringSearchModelInterpolatorTest.java
+++ b/maven-model-builder/src/test/java/org/apache/maven/model/interpolation/StringSearchModelInterpolatorTest.java
@@ -36,8 +36,6 @@ import java.util.concurrent.FutureTask;
 import static org.hamcrest.CoreMatchers.anyOf;
 import static org.hamcrest.CoreMatchers.is;
 import static org.junit.Assert.assertThat;
-import static org.powermock.reflect.Whitebox.getField;
-import static org.powermock.reflect.Whitebox.getInternalState;
 
 /**
  * @author jdcasey
@@ -375,70 +373,6 @@ public class StringSearchModelInterpolatorTest
         ) ) );
     }
 
-    public void testNotInterpolateObjectWithFile()
-            throws Exception
-    {
-        Model model = new Model();
-
-        File baseDir = new File( System.getProperty( "user.dir" ) );
-
-        Properties p = new Properties();
-
-        ObjectWithNotInterpolatedFile obj = new ObjectWithNotInterpolatedFile( baseDir );
-
-        StringSearchModelInterpolator interpolator = (StringSearchModelInterpolator) createInterpolator();
-
-        ModelBuildingRequest config = createModelBuildingRequest( p );
-
-        SimpleProblemCollector collector = new SimpleProblemCollector();
-        interpolator.interpolateObject( obj, model, new File( "." ), config, collector );
-        assertProblemFree( collector );
-
-        //noinspection unchecked
-        Map<Class<?>, ?> cache =
-                (Map<Class<?>, ?>) getField( StringSearchModelInterpolator.class, "CACHED_ENTRIES" )
-                        .get( null );
-
-        Object objCacheItem = cache.get( Object.class );
-        Object fileCacheItem = cache.get( File.class );
-
-        assertNotNull( objCacheItem );
-        assertNotNull( fileCacheItem );
-
-        assertThat( ( (Object[]) getInternalState( objCacheItem, "fields" ) ).length, is( 0 ) );
-        assertThat( ( (Object[]) getInternalState( fileCacheItem, "fields" ) ).length, is( 0 ) );
-    }
-
-    public void testNotInterpolateFile()
-            throws Exception
-    {
-        Model model = new Model();
-
-        File baseDir = new File( System.getProperty( "user.dir" ) );
-
-        Properties p = new Properties();
-
-        StringSearchModelInterpolator interpolator = (StringSearchModelInterpolator) createInterpolator();
-
-        ModelBuildingRequest config = createModelBuildingRequest( p );
-
-        SimpleProblemCollector collector = new SimpleProblemCollector();
-        interpolator.interpolateObject( baseDir, model, new File( "." ), config, collector );
-        assertProblemFree( collector );
-
-        //noinspection unchecked
-        Map<Class<?>, ?> cache =
-                (Map<Class<?>, ?>) getField( StringSearchModelInterpolator.class, "CACHED_ENTRIES" )
-                        .get( null );
-
-        Object fileCacheItem = cache.get( File.class );
-
-        assertNotNull( fileCacheItem );
-
-        assertThat( ( (Object[]) getInternalState( fileCacheItem, "fields" ) ).length, is( 0 ) );
-    }
-
-
     public void testConcurrentInterpolation()
         throws Exception
     {
-- 
2.25.1

