From 39a39fb72dc9d392fc9a4ccf5e2f96e0d35cfb09 Mon Sep 17 00:00:00 2001
From: mbykhovtsev-ms <mbykhovtsev@microsoft.com>
Date: Thu, 23 Mar 2023 15:39:16 -0700
Subject: [PATCH 6/6] Restore DefaultModelValidator compatibility with Maven 3.5.4

---
 .../maven/model/validation/DefaultModelValidator.java       | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java b/maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java
index 9f11fa2..e9cb37e 100644
--- a/maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java
+++ b/maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java
@@ -40,6 +40,7 @@ import org.apache.maven.model.Reporting;
 import org.apache.maven.model.Repository;
 import org.apache.maven.model.Resource;
 import org.apache.maven.model.building.ModelBuildingRequest;
+import org.apache.maven.model.interpolation.DefaultModelVersionProcessor;
 import org.apache.maven.model.building.ModelProblem.Severity;
 import org.apache.maven.model.building.ModelProblem.Version;
 import org.apache.maven.model.building.ModelProblemCollector;
@@ -85,6 +86,11 @@ public class DefaultModelValidator
 
     private ModelVersionProcessor versionProcessor;
 
+    public DefaultModelValidator()
+    {
+        this( new DefaultModelVersionProcessor() );
+    }
+
     @Inject
     public DefaultModelValidator(  ModelVersionProcessor versionProcessor )
     {
-- 
2.25.1

